docs/using/libraries/rust/index.rst

```diff
@@ -0,0 +1,175 @@
+============================
+Rust Authorization Library
+============================
+
+oso is packaged as a :doc:`cargo</download>` crate for use in Rust applications.
+
+API documentation for the crate lives `on docs.rs <https://docs.rs/oso/>`_.
+
+
+.. toctree::
+    :hidden:
+
+To install, see :doc:`installation instructions </download>`.
+
+Working with Rust Types
+===========================
+
+oso's Rust authorization library allows you to write policy rules over Rust types directly.
+This document explains how different Rust types can be used in oso policies.
+
+.. note::
+  More detailed examples of working with application objects can be found in :doc:`/using/examples/index`.
+
+Structs + Enums
+^^^^^^^^^^^^^^^
+
+Rust structs and enums can be registered with oso which lets you pass them in and access their methods and fields. (see :ref:`application-types`).
+
+Rust structs can also be constructed from inside an oso policy using the :ref:`operator-new` operator if the type has been given a constructor when registered.
+
+Numbers and Booleans
+^^^^^^^^^^^^^^^^^^^^
+Polar supports both integer and floating point numbers, as well as booleans (see :ref:`basic-types`).
+
+Strings
+^^^^^^^
+Rust `Strings <https://doc.rust-lang.org/std/string/struct.String.html>`_ are mapped to Polar :ref:`strings`. Many of rust's string methods may be called in policies:
+
+.. code-block:: polar
+  :caption: :fa:`oso` policy.polar
+
+  allow(actor, action, resource) if actor.username.ends_with("example.com");
+
+.. code-block:: rust
+  :caption: :fab:`rust` main.rs
+
+  #[derive(Clone, PolarClass)]
+  struct User {
+    #[polar(attribute)]
+    pub username: String
+  }
+
+  oso.register_class(User::get_polar_class())?;
+
+  let user = User{username: "alice@example.com".to_owned()};
+  assert!(oso.is_allowed(user, "foo", "bar")?);
+
+.. warning::
+  Polar does not support methods that mutate strings in place.
+
+
+Vectors
+^^^^^^^
+`Vec<T> <https://doc.rust-lang.org/std/vec/struct.Vec.html>`_ is mapped to Polar :ref:`Lists <lists>`, given that ``T: ToPolar``. 
+
+Currently, no methods on ``Vec`` are exposed to Polar.
```

Comment:
Do we expose any list methods in other languages? If so we should follow up on this.

Reply:
We do, we expose them all, and have parity tests using them

Reply:
Aight, it goes on the list then, maybe even all the iterator methods 😨

Reply:
Let's just make sure people are able to add their own methods :) we can provide the basics.
Who needs iterator methods when you have Polar? :)